apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/questionnaire/QuestionnaireMapper.java

```diff
@@ -0,0 +1,123 @@
+import gov.cdc.nbs.questionbank.entities.DisplayElementGroupEntity;
+import gov.cdc.nbs.questionbank.entities.DisplayElementRef;
+import gov.cdc.nbs.questionbank.entities.DisplayGroupRef;
+import gov.cdc.nbs.questionbank.entities.DropDownQuestionEntity;
+import gov.cdc.nbs.questionbank.entities.NumericQuestionEntity;
+import gov.cdc.nbs.questionbank.entities.QuestionnaireEntity;
+import gov.cdc.nbs.questionbank.entities.Reference;
+import gov.cdc.nbs.questionbank.entities.TextEntity;
+import gov.cdc.nbs.questionbank.entities.TextQuestionEntity;
+import gov.cdc.nbs.questionbank.entities.ValueEntity;
+import gov.cdc.nbs.questionbank.entities.ValueSet;
+import gov.cdc.nbs.questionbank.questionnaire.model.Questionnaire;
+
+class QuestionnaireMapper {
+
+    /**
+     * Converts a {@link gov.cdc.nbs.questionbank.entities.QuestionnaireEntity QuestionnaireEntity} into a
+     * {@link gov.cdc.nbs.questionbank.questionnaire.model.Questionnaire Questionnaire}.
+     * 
+     * @param optional
+     * @return
+     */
+    public Questionnaire toQuestionnaire(QuestionnaireEntity entity) {
+        return new Questionnaire(
+                entity.getId(),
+                entity.getConditionCodes(),
+                entity.getReferences().stream().map(this::toElement).toList());
+    }
+
+    /**
+     * Accepts a {@link gov.cdc.nbs.questionbank.entities.Reference Reference} and converts it into a
+     * {@link gov.cdc.nbs.questionbank.questionnaire.model.Questionnaire.Element Questionnaire.Element}.
+     * 
+     * A reference can be either a DisplayElementRef that references a
+     * {@link gov.cdc.nbs.questionbank.entities.DisplayElementEntity DisplayElementEntity} or a DisplayElementGroupRef
+     * that references a {@link gov.cdc.nbs.questionbank.entities.DisplayElementGroupEntity DisplayElementGroupEntity}.
+     * 
+     * @param ref
+     * @return
+     */
+    Questionnaire.Element toElement(Reference ref) {
+        if (ref instanceof DisplayElementRef displayElementRef) {
+            return toDisplayElement(displayElementRef.getDisplayElementEntity());
+        } else if (ref instanceof DisplayGroupRef displayGroupRef) {
+            return toSection(displayGroupRef.getDisplayGroup());
+        } else {
+            throw new IllegalArgumentException("Unsupported display element reference type: " + ref.getReferenceType());
+        }
+    }
+
+    /**
+     * Converts a {@link gov.cdc.nbs.questionbank.entities.DisplayElementEntity DisplayElementEntity} to its
+     * corresponding {@link gov.cdc.nbs.questionbank.questionnaire.model.Questionnaire.Element Questionnaire.Element}.
+     * 
+     * @param element
+     * @return
+     */
+    Questionnaire.DisplayElement toDisplayElement(DisplayElementEntity element) {
+        if (element instanceof TextEntity t) {
+            return new Questionnaire.Text(t.getId(), t.getText());
+        } else if (element instanceof TextQuestionEntity tq) {
```

Comment:
could potentially be converted to switch case

Reply:
Tried that with a couple approaches:
![image](https://github.com/CDCgov/NEDSS-Modernization/assets/109251240/fd7a1fc2-8738-45b1-8122-fd70ba764d66)
This works and we could possibly make the "type" field public on the different entities as to not use "text", but its a bit longer than the if statement and involves doing the class casting separately.

Ideally we would use the new pattern matching for switch statements, but it is not released as of java 17

https://www.atlantbh.com/pattern-matching-for-instanceof-and-switch-in-java/


If I overlooked a better approach let me know

Reply:
No that is fine if it will make it unnecessarily longer and more complex your current solution is the better option. And  yeah that new pattern matching for switch statements feature currently in preview looks cool and will definitely be helpful.